R/mod_tableGen_utils.R

```diff
@@ -445,6 +445,38 @@ pretty_IDs <- function(ID) {
   )
 }
 
+prep_blocks <- function(blockData) {
+  dput(blockData) %>%
+    capture.output() %>%
+    paste0(collapse = "") %>%
+    str_replace_all("\\s{2,}", " ") %>%
+    str_replace_all("(\\),)", "\\1\n")
+}
+
+std_footnote <- function(data, source) {
+  gt::tab_footnote(data, 
+                   tags$div(HTML("<b>Source:</b>", source), 
+                            shiny::tags$span(shiny::HTML("<b> Run Date:</b>", toupper(format(Sys.Date(), "%d%b%Y"))),
+                                             style="float:right"),
+                            style="text-align:left"))
+}
+
+tg_gt <- function(tg_datalist, blockData, total_df, group) {
+  purrr::pmap(list(
+    blockData$agg,
+    blockData$S3,
+    blockData$dropdown,
+    blockData$dataset), 
+    function(x,y,z,d) tidyCDISC::app_methods(x,y,z,
+                                             group = group, 
+                                             data = tidyCDISC::data_to_use_str(d, tg_datalist$ADAE, tg_datalist$ADSL),
+                                             totals = total_df)) %>%
+    purrr::map(setNames, tidyCDISC::common_rownames(tg_datalist$POPDAT, group)) %>%
+    setNames(paste(blockData$gt_group)) %>%
+    dplyr::bind_rows(.id = 'ID') %>%
+    dplyr::mutate(ID = tidyCDISC::pretty_IDs(ID))
+}
+
```

Comment:
These new functions will need some `{roxygen2}` code to document and export where needed, plus will need some `{testthat}` tests!

Reply:
This is done in part. Examples and tests still need to be created, but appropriate functions have been exported.